SU2_CFD/include/gradients/correctGradientsSymmetry.hpp

```diff
@@ -61,17 +61,25 @@ inline void correctGradient(const size_t varBegin, const size_t varEnd, const in
       }
     }
 
-    /*--- Normal gradient of the normal velocity. ---*/
-    const su2double normalGradNormalVel = GeometryToolbox::DotProduct(nDim, n, gradNormalVel);
+    // /*--- Normal gradient of the normal velocity. ---*/
+    // const su2double normalGradNormalVel = GeometryToolbox::DotProduct(nDim, n, gradNormalVel);
 
-    /*--- Remove the tangential projection (I - n.n^T) of the normal gradients.
-     * And the normal projection (n.n^T) of the tangential gradients.
-     * dV = dV - (I - n.n^T) dV n.n^T - n.n^T dV (I - n.n^T) ---*/
+    // /*--- Remove the tangential projection (I - n.n^T) of the normal gradients.
+    //  * And the normal projection (n.n^T) of the tangential gradients.
+    //  * dV = dV - (I - n.n^T) dV n.n^T - n.n^T dV (I - n.n^T) ---*/
 
+    // for (size_t iDim = 0; iDim < nDim; iDim++) {
+    //   for (size_t jDim = 0; jDim < nDim; jDim++) {
+    //     gradients[idxVel + iDim][jDim] -= normalGrad[iDim] * n[jDim] + n[iDim] * gradNormalVel[jDim];
+    //     gradients[idxVel + iDim][jDim] += 2 * n[iDim] * normalGradNormalVel * n[jDim];
+    //   }
+    // }
+
+    // Do I really need all of the previous code? Can't I just merge the two for loops? Maybe I am missing the tangential direction
```

Comment:
Yes what you are doing neglects that velocity is a vector quantity

Reply:
The previous implementation was actually correct, I'll revert to that one. Nevertheless something is off.